mlflow/metrics/genai/model_utils.py

```diff
@@ -9,9 +10,32 @@
 _logger = logging.getLogger(__name__)
 
 
+def get_endpoint_type(endpoint_uri: str) -> Optional[str]:
+    """
+    Get the type of the endpoint if it is MLflow deployment
+    endpoint. For other endpoints e.g. OpenAI, or if the
+    endpoint does not specify type, return None.
+    """
+    schema, _ = _parse_model_uri(endpoint_uri)
+
+    if schema != "endpoints":
+        return None
+
+    from pydantic import BaseModel
+
+    from mlflow.deployments import get_deploy_client
+
+    client = get_deploy_client()
+
+    endpoint = client.get_endpoint(endpoint_uri)
+    # TODO: Standardize the return type of `get_endpoint` and remove this check
+    endpoint = endpoint.dict() if isinstance(endpoint, BaseModel) else endpoint
+    return endpoint.get("task", endpoint.get("endpoint_type"))
+
+
 # TODO: improve this name
-def score_model_on_payload(model_uri, payload, eval_parameters=None):
-    """Call the model identified by the given uri with the given payload."""
+def score_model_on_payload(model_uri, payload, eval_parameters=None, endpoint_type=None):
```

Comment:
note: I will file a follow-up to rename `payload` to `prompt` (and elsewhere in the caller side). The `payload` is over-loaded within genai metrics code base, it is sometimes string (`content`) while also refers to the actual chat payload. This is very confusing.

Reply:
Agreed on this point. Far too generic of a term when reading through these APIs :)